File: javascript/1122.Relative-Sort-Array.js
/**
 * @param {number[]} arr1
 * @param {number[]} arr2
 * @return {number[]}
 */
var relativeSortArray = function (arr1, arr2) {
  const rank = new Map();

  arr2.forEach((num, i) => rank.set(num, i));

  arr1.sort((a, b) => {
    const ra = rank.has(a) ? rank.get(a) : Infinity;
    const rb = rank.has(b) ? rank.get(b) : Infinity;

    return ra === rb ? a - b : ra - rb;
  });

  return arr1;
};

File: python/1122.Relative-Sort-Array.py
class Solution:
    def relativeSortArray(self, arr1: List[int], arr2: List[int]) -> List[int]:
        freq = Counter(arr1)
        res = []

        for n in arr2:
            if n in freq:
                res.extend([n] * freq[n])
                del freq[n]
        
        for n in sorted(freq):
            res.extend([n] * freq[n])

        return res